### ISSUE TITLE
[Java] Implement new Concept Exercise: interfaces

### ISSUE THREAD

**Issue by mirkoperillo** [exercism/v3#1870]
This issue describes how to implement the `interfaces` concept exercise for the Java track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal of this exercise is to teach the student the basics of the Concept of Interfaces in Java.

## Learning objectives

- Know what interfaces are.
- Know how to use interfaces.
- Know how to define an interface.
- Know how to implement an interface.

## Out of scope

- Default interface methods.

## Concepts

The Concepts this exercise unlocks are:

- `interfaces`: know what interfaces are; know how to use interfaces; know how to define an interface; know how to implement an interface.

## Prequisites

This exercise's prerequisites Concepts are:

- `classes`: know how to work with classes.

### Hints

- [Interface][interface]: what an interface is and how to implement it.

### After

- [Interface][interface]: what an interface is and how to implement it.

## Representer

This exercise does not require any specific representation logic to be added to the [representer][representer].

## Analyzer

This exercise does not require any specific analyzer logic to be added to the [analyzer][analyzer].

## Implementing

To implement this exercise, please [follow these instructions](https://github.com/exercism/v3/blob/master/languages/java/reference/implementing-a-concept-exercise.md).

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

[analyzer]: https://github.com/exercism/java-analyzer
[representer]: https://github.com/exercism/java-representer
[interface]: https://docs.oracle.com/javase/tutorial/java/concepts/interface.html

**Comment by mikedamay** [comment]
Closed via PR #2248
